flink-runtime/src/test/java/org/apache/flink/runtime/source/coordinator/SourceCoordinatorAlignmentTest.java

```diff
@@ -131,6 +133,34 @@ void testWatermarkAlignmentWithTwoGroups() throws Exception {
         }
     }
 
+    @Test
+    @Timeout(value = 10)
+    void testSendWatermarkAlignmentEventFailed() throws Exception {
+        long maxDrift = 1000L;
+
+        WatermarkAlignmentParams params =
+                new WatermarkAlignmentParams(maxDrift, "group1", maxDrift);
+        sourceCoordinator = getNewSourceCoordinator(params);
+        sourceCoordinator.start();
+
+        final int subtask = 0;
+        int attemptNumber = 0;
+        sourceCoordinator.handleEventFromOperator(
+                subtask,
+                attemptNumber,
+                new ReaderRegistrationEvent(subtask, createLocationFor(subtask, attemptNumber)));
+        // SubTask ReportedWatermarkEvent before setReaderTaskReady to simulate task failover
+        sourceCoordinator.handleEventFromOperator(
+                subtask, attemptNumber, new ReportedWatermarkEvent(1000));
+        Thread.sleep(3000);
```

Comment:
This should be converted to a test without busy waiting and a timeout. I'm not sure what are you waiting for, but this should be changed to some latch, or other waiting condition.